### PR TITLE
Deminify 

### PR DESCRIPTION
**Pull request by grafixeyehero**

### CODE REVIEW

**Comment by grafixeyehero** [comment]
Update Suggested change done

File: src/scripts/livetvcomponents.js [hunk 1]
```diff
@@ -1,73 +1,115 @@
...
+
+            if (false !== options.indexByDate && item.StartDate) {
+                try {
+                    var premiereDate = datetime.parseISO8601Date(item.StartDate, true);
+                    dateText = datetime.toLocaleDateString(premiereDate, {
+                        weekday: "long",
+                        month: "short",
+                        day: "numeric"
+                    });
+                } catch (err) {}
```
[the remark this answers is not in the document]
Comment (grafixeyehero) [reply src/scripts/livetvcomponents.js]: Done https://github.com/jellyfin/jellyfin-web/pull/506/commits/6eb8e1b8fa8fb2ab28647cb3f8fbde80c7c4fb74

File: src/scripts/taskbutton.js [hunk 2]
```diff
@@ -1,48 +1,103 @@
...
+
+                if (pollInterval) {
+                    clearInterval(pollInterval);
+                }
+            })();
+        } else {
+            button.addEventListener("click", onButtonClick);
+            pollTasks();
+
+            (function () {
```
[the remark this answers is not in the document]
Comment (grafixeyehero) [reply src/scripts/taskbutton.js]: Done https://github.com/jellyfin/jellyfin-web/pull/506/commits/0e306cad76021063affb75f65af606ec0f2ed7fb

File: src/scripts/livetvcomponents.js [hunk 3]
```diff
@@ -1,73 +1,115 @@
...
+define(["layoutManager", "datetime", "cardBuilder", "apphost"], function (layoutManager, datetime, cardBuilder, appHost) {
     "use strict";
 
     function enableScrollX() {
-        return !layoutManager.desktop
+        return !layoutManager.desktop;
     }
 
     function getBackdropShape() {
-        return enableScrollX() ? "overflowBackdrop" : "backdrop"
```
[the remark this answers is not in the document]
Comment (grafixeyehero) [reply src/scripts/livetvcomponents.js]: Done https://github.com/jellyfin/jellyfin-web/pull/506/commits/e7570d856fd214d5805291fcc5498763bc1e08f9

File: src/scripts/itembynamedetailpage.js [hunk 4]
```diff
@@ -190,57 +248,132 @@ define(["connectionManager", "listView", "cardBuilder", "imageLoader", "libraryB
...
+        }
+
+        return "list.html?type=" + type + "&parentId=" + item.Id + "&serverId=" + item.ServerId;
     }
 
     function addCurrentItemToQuery(query, item) {
-        "Person" == item.Type ? query.PersonIds = item.Id : "Genre" == item.Type ? query.GenreIds = item.Id : "MusicGenre" == item.Type ? query.GenreIds = item.Id : "Studio" == item.Type ? query.StudioIds = item.Id : "MusicArtist" == item.Type && (connectionManager.getApiClient(item.ServerId).isMinServerVersion("3.4.1.18") ? query.AlbumArtistIds = item.Id : query.ArtistIds = item.Id)
+        if ("Person" == item.Type) {
+            query.PersonIds = item.Id;
+        } else {
```
[the remark this answers is not in the document]
Comment (grafixeyehero) [reply src/scripts/itembynamedetailpage.js]: Done https://github.com/jellyfin/jellyfin-web/pull/506/commits/e7570d856fd214d5805291fcc5498763bc1e08f9

File: src/scripts/itembynamedetailpage.js [hunk 5]
```diff
@@ -190,57 +248,126 @@ define(["connectionManager", "listView", "cardBuilder", "imageLoader", "libraryB
...
+        } else if (item.Type == "GameGenre") {
+            query.Genres = item.Name;
+        } else if (item.Type == "Studio") {
+            query.StudioIds = item.Id;
+        } else if (item.Type == "MusicArtist") {
+            if (connectionManager.getApiClient(item.ServerId).isMinServerVersion("3.4.1.18")) {
+                query.AlbumArtistIds = item.Id;
+            } else {
+                query.ArtistIds = item.Id;
+            }
```
Comment (JustAMan) [inline src/scripts/itembynamedetailpage.js]: We're on version 3.5 anyway, so let's simplify it to
```js
query.AlbumArtistIds = item.Id;
```

File: src/scripts/librarymenu.js [hunk 6]
```diff
@@ -198,18 +199,22 @@ define(["dom", "layoutManager", "inputManager", "connectionManager", "events", "
...
+
             html += '<a is="emby-linkbutton" class="navMenuOption lnkMediaFolder btnLogout" data-itemid="logout" href="#"><i class="md-icon navMenuOptionIcon">exit_to_app</i><span class="navMenuOptionText">' + globalize.translate("ButtonSignOut") + "</span></a>";
             html += "</div>";
         }
 
         // add buttons to navigation drawer
-        navDrawerScrollContainer.innerHTML = html;
 
+        navDrawerScrollContainer.innerHTML = html;
         // bind logout button click to method
```
Comment (JustAMan) [inline src/scripts/librarymenu.js]: move it one line down pls

File: src/scripts/librarymenu.js [hunk 7]
```diff
@@ -198,18 +199,22 @@ define(["dom", "layoutManager", "inputManager", "connectionManager", "events", "
...
+
             if (appHost.supports("multiserver")) {
                 html += '<a is="emby-linkbutton" class="navMenuOption lnkMediaFolder" data-itemid="selectserver" href="selectserver.html?showuser=1"><i class="md-icon navMenuOptionIcon">wifi</i><span class="navMenuOptionText">' + globalize.translate("ButtonSelectServer") + "</span></a>";
             }
+
             html += '<a is="emby-linkbutton" class="navMenuOption lnkMediaFolder btnLogout" data-itemid="logout" href="#"><i class="md-icon navMenuOptionIcon">exit_to_app</i><span class="navMenuOptionText">' + globalize.translate("ButtonSignOut") + "</span></a>";
             html += "</div>";
         }
 
         // add buttons to navigation drawer
```
Comment (JustAMan) [inline src/scripts/librarymenu.js]: move it one line down pls, comment should not be separated by an empty line from the code it comments IMHO

File: src/scripts/librarymenu.js [hunk 8]
```diff
@@ -501,17 +506,16 @@ define(["dom", "layoutManager", "inputManager", "connectionManager", "events", "
         }
 
         // FIXME: Potentially the same as above
+
```
Comment (JustAMan) [inline src/scripts/librarymenu.js]: I think this empty line isn't needed

File: src/scripts/livetvcomponents.js [hunk 9]
```diff
@@ -1,73 +1,118 @@
...
+            if (options.indexByDate !== false && item.StartDate) {
+                try {
+                    var premiereDate = datetime.parseISO8601Date(item.StartDate, true);
+                    dateText = datetime.toLocaleDateString(premiereDate, {
+                        weekday: "long",
+                        month: "short",
+                        day: "numeric"
+                    });
+                } catch (err) {
+                    console.log("Error parsing premiereDate:" + item.StartDate);
```
Comment (JustAMan) [inline src/scripts/livetvcomponents.js]:
```suggestion
                    console.log("Error parsing premiereDate:" + item.StartDate + "; error: " + err);
```

Comment (grafixeyehero) [reply src/scripts/livetvcomponents.js]: https://github.com/jellyfin/jellyfin-web/pull/506/commits/af3579f328781247f008fa5f05940993fae945fa

File: src/scripts/itembynamedetailpage.js (hunk 5 above)
Comment (grafixeyehero) [reply src/scripts/itembynamedetailpage.js]: https://github.com/jellyfin/jellyfin-web/pull/506/commits/03a16287a033c45812fe5ecee444d1b98fb207c3

**Pull request merged**